### DISCUSSION TITLE
[Roadmap]Superset proposed feature roadmap 2021-2022

### DISCUSSION THREAD

**Discussion by junlincc** [apache/superset#13883] [Show and tell]
Dear Superset users, 

Thanks to all your support, feedback, and patience, we successfully launched Superset v1.0 last month. As our development team is working hard on polishing the product to provide the best user experience, we want to think ahead of the future and next milestone of Superset. 

We put together some proposed feature roadmap items in (https://docs.google.com/spreadsheets/d/1V82lxvWeWJK48o8S0OkeTnlL409DbtZSa9qTzdSHx6I/edit#gid=0) for Superset 2021-2022 development. It's now available for everyone to review and comment. 

If ☝️that's too much to digest, we also set up a feature upvote page for you to help us prioritize.
 ----> https://superset.hellonext.co/ 
Please let us know the most important features to you and your organization by upvoting them. 
Note: the upvote list is only a subset of the complete proposed list. It does not include projects already in development. 

<img width="672" alt="Screen Shot 2021-03-05 at 10 20 30 AM" src="https://user-images.githubusercontent.com/67837651/110157032-913d0e80-7d9c-11eb-83e8-2ba8c9f71533.png">

<img width="1214" alt="Screen Shot 2021-02-25 at 11 40 43 PM" src="https://user-images.githubusercontent.com/67837651/109272503-cc9c7380-77c5-11eb-8497-152a0e82028e.png">
<img width="1655" alt="Screen Shot 2021-02-25 at 11 41 09 PM" src="https://user-images.githubusercontent.com/67837651/109272523-d1f9be00-77c5-11eb-917e-edf845f9e1e7.png">

The steps we took 
- Scape entire repo issues, mark issues that are feature requests
- Categorize the ones that are aligned with Superset long-term product vision, mark committed
- Classify all the committed items by user persona and product area with a defined theme and goal  
- Set priority based on items' impact level, motivation, risk, project size, and type 
- Open the discussion to the Superset development team(PMC and committer) to collect feedback 
- Adjust based on the first round of feedback 
- Open the discussion to bigger Superset community (Current stage)

We will keep the google sheet and upvote list open for discussion for a week before iterating on them.
-  If the feature you want is missing from the lists
-  If you want to take on ownership or contribute to any of these items
-  If you disagree with any details stated in the sheet 
please feel free to leave your comment here ⬇ 

Thanks again for choosing Superset as your BI tool~ 😉

**Comment by vpicavet** [comment]
Hi, thanks for openly providing your roadmap. It looks great !

I cannot find anywhere our number 1 missing feature : a widget of templated markdown text, where values are inserted from a SQL query. "big number" widget is good, but we need more dense information widget and we love markdown, but it has to be dynamic !

Sidenote : hellonext requires read AND write access to github repos, as well as full personal info access, which is a big NO-NO !

**Comment by rusackas** [comment]
> Sidenote : hellonext requires read AND write access to github repos, as well as full personal info access, which is a big NO-NO !

Thanks for you input and reasonable concerns. There are alternate authentication methods for HelloNext if you'd rather not use Github auth. HelloNext voting is also quite optional. If you want to side-step HelloNext, I don't see why you couldn't make a bullet list of `+1`s and paste it here.

We're just trying to get some feedback/input here, this is not the roadmap driver of software-by-vote process.

**Comment by junlincc** [comment]
> I cannot find anywhere our number 1 missing feature : a widget of templated markdown text, where values are inserted from a SQL query. "big number" widget is good, but we need more dense information widget and we love markdown, but it has to be dynamic !
> 
Hi @vpicavet, thanks for requesting feature. Would you mind sharing a link to the your request if you filed one before? I tried to search but did not find anything. 

There are a few reasons why any particular item are not included in the roadmap item list. 
This feature has not been highly requested by the community 2) The use case and values are not clear 3) It may not perfectly be aligned with the product vision.
If this feature is important to you and your org, please do provide more info. Now your comment is in public. Hopefully, it will gain more visibility for people to vote it up by replying to the thread.

> Sidenote : hellonext requires read AND write access to github repos, as well as full personal info access, which is a big NO-NO !

As @rusackas mentioned,  Connecting to the user's Github account is only one way to sign-in hellonext, the user can simply comment in the roadmap sheet or here to provide feedback. From the product standpoint, we need to ensure each vote's quality and uniqueness, which helps us evaluate the impact of different roadmap items.
Thanks again for bringing up your request and concern.

Thanks again for bringing up your request and concern.

**Comment by vpicavet** [comment]
Hello, thanks for your welcoming answers.

I did not filed a feature request yet, as I found the amount of issues filed recently quite overwhelming already, and was not sure something alike had not been filed before. But I will create an issue and try to be as descriptive as possible.

I added some comments on the Google sheet. I have no trouble with authentication for voters, but requiring write acess to repos and full personal info from hellonext sounds clearly like an abuse of power. As soon as there are other methods, it does not really matter, but I was surprised and wanted to let you know.

I will post the link to the feature request as soon as I fill it.

**Comment by vpicavet** [comment]
I checked through issues history and my need would be aligned with already proposed features, such as reviving the Markup Viz chart, or using Jinja2 templates, or a parameterized Markdown chart.

Even if it would prevent inserting dynamic images, something like a "Big card" chart would already be great, as I could generate the full text in SQL. Less convenient and no image though.

References : 
- Big card : https://github.com/apache/superset/issues/12516
- Jinja2 support : https://github.com/apache/superset/issues/6771
- Description of a totally similar need : https://github.com/apache/superset/issues/5747

**Comment by junlincc** [comment]
@vpicavet thanks for the additional information! we will reevaluate the request, and update the roadmap if needed. 🙂

**Comment by raags** [comment]
@junlincc was this request evaluated, and is it part of the 2022 roadmap?

edit: I can see https://github.com/apache/superset/pull/17903 is close to merging, and it looks like it'll address my needs.
